edx_sga/sga.py

```diff
@@ -508,20 +513,36 @@ def staff_download_annotated(self, request, suffix=''):
         return self.download(
             path,
             state['annotated_mimetype'],
-            state['annotated_filename']
+            state['annotated_filename'],
+            require_staff=True
         )
 
-    def download(self, path, mime_type, filename):
+    def download(self, path, mime_type, filename, require_staff=False):
         """
         Return a file from storage and return in a Response.
         """
-        file_descriptor = default_storage.open(path)
-        app_iter = iter(partial(file_descriptor.read, BLOCK_SIZE), '')
-        return Response(
-            app_iter=app_iter,
-            content_type=mime_type,
-            content_disposition=("attachment; filename=" +
-                                 filename.encode('utf-8')))
+        try:
+            file_descriptor = default_storage.open(path)
+            app_iter = iter(partial(file_descriptor.read, BLOCK_SIZE), '')
+            return Response(
+                app_iter=app_iter,
+                content_type=mime_type,
+                content_disposition="attachment; filename=" + filename.encode('utf-8'))
+        except IOError:
+            if require_staff:
```

Comment:
Why is this conditional here? Students can download their own uploads, so there is the potential of an error for them as well.